waveform_editor/shape_editor/nice_plotter.py

```diff
@@ -0,0 +1,219 @@
+import holoviews as hv
+import matplotlib.pyplot as plt
+import panel as pn
+import param
+
+from waveform_editor.shape_editor.nice_integration import NiceIntegration
+
+
+class NicePlotter(param.Parameterized):
+    communicator = param.ClassSelector(class_=NiceIntegration)
+
+    DEFAULT_OPTS = {
+        "xlim": (0, 13),
+        "ylim": (-10, 10),
+        "title": "Equilibrium poloidal flux",
+        "xlabel": "r [m]",
+        "ylabel": "z [m]",
+    }
+    COLORBAR_OPTS = {"title": "ψ [Wb]"}
+    WIDTH = 800
+    HEIGHT = 1000
+
+    def __init__(self, communicator, plotting_params, **params):
+        super().__init__(**params)
+        self.communicator = communicator
+        self.plotting_params = plotting_params
+        self.wall = None
+        self.pf_active = None
+
+    @pn.depends("plotting_params.param", "communicator.processing")
```

Comment:
Would it be possible to construct several DynamicMaps for the different components?

Right now there is a ~second delay to just turn on/off display of e.g. coils (probably because it's recalculating all of the contours).

Reply:
As discussed, see #95